File: src/esp32.WebApi.Abstraction/DTO/ProdutoDTO.cs
using System;
using System.Collections.Generic;

namespace esp32.WebApi.Abstraction.DTO {
    public partial class ProdutoDTO
    {
        public Guid Idproduto { get; set; }
        public string Nome { get; set; }
        public string Marca { get; set; }
        public float Peso { get; set; }
        public DateTime? Inativo { get; set; }

        public virtual BalancaDTO[] Balanca { get; set; }
    }
}

File: src/UI/esp32-ui/src/app/shared/DTOs/ProdutoDTO.ts
import { BalancaDTO } from './BalancaDTO';
import { HistoricoProdutoDTO } from './HistoricoProdutoDTO';

export class ProdutoDTO {
  idproduto: string;
  nome: string;
  marca: string;
  peso: number;
  inativo?: Date;

  balanca: BalancaDTO[];
  historicoProduto: HistoricoProdutoDTO[];
}
